server/service/apple_mdm.go

```diff
@@ -2113,6 +2134,9 @@ func ReconcileProfiles(
 	//
 	// We'll do another pass at the end to revert any changes for failed
 	// delivieries.
+	if err := ds.BulkDeleteMDMAppleHostProfiles(ctx, replacedHostProfiles); err != nil {
```

Comment:
My thinking here was that if the install fails, there would be an entry for the install failure, which would tie back to the profile identifier of the to-be-replaced profile. I think that is consistent with what the UI wants to display, but do you think we need more than that entry for our own table hygiene?

Reply:
Ah yeah, you're right, the profile to be removed would be deleted from the table, but the one to be inserted as replacement would also be inserted and with a status NULL so that enqueuing the Install command is retried on the next run. It would be reported in stats/UI/etc. as "pending install" which is exactly what we want.